### PR TITLE
Add CLI functionality and separate out some logic

### PR DESCRIPTION
**Pull request by mlieberman85**
This adds the CLI, and also separates out intoto attestation logic from the SLSA predicate.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,14 +1,21 @@
...
+cargo run validate in-toto-v1 slsa-provenance-v1 --file tests/fixtures/slsa_provenance_v1.json
+```
+
+You can replace the `slsa_provenance_v1.json` with another in-toto statement and even an invalid one to verify the correctness of the document. 
+
+## Developing and Building
 Spector is written in Rust, and built with [cargo](https://doc.rust-lang.org/book/ch01-03-hello-cargo.html)
-Check out the code and run `cargo build`
\ No newline at end of file
+Check out the code and run `cargo build` or `cargo test`.
```
Comment (06kellyjac) [inline README.md:21]: No newline at end of file

File: src/bin/bin.rs [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+    version = "0.0.1",
+    about = "A tool for validating supply chain metadata documents"
+)]
+struct Spector {
+    #[clap(subcommand)]
+    command: Command,
+}
+
+/// The available subcommands.
+/// Currently only supports validating documents
```
Comment (06kellyjac) [inline src/bin/bin.rs:24]:
```suggestion
```

Covered below

File: tests/fixtures/slsa_provenance_v1_invalid_predicate.json [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    },
+    "subject": [
+        {
+            "name": "_",
+            "digest": {
+                "sha256": "fe4fe40ac7250263c5dbe1cf3138912f3f416140aa248637a60d65fe22c47da4"
+            }
+        }
+    ]
+}
```
Comment (06kellyjac) [inline tests/fixtures/slsa_provenance_v1_invalid_predicate.json:58]: Newline

**Review by pxp928 — APPROVED** [review]

**Review by pxp928 — APPROVED** [review]
LGTM! Great tests!

**Pull request merged**